tests/python/gaia-ui-tests/gaiatest/apps/ftu/app.py

```diff
@@ -159,8 +159,14 @@ def tap_next_to_wifi_section(self):
 
     def a11y_click_next_to_wifi_section(self):
         self.a11y_click_next()
-        self.wait_for_element_not_displayed(*self._loading_overlay_locator)
-        self.wait_for_element_displayed(*self._section_wifi_locator)
+        try:
+            Wait(self.marionette, timeout, ignored_exceptions=StaleElementException).until(
+                lambda m: not m.find_element(*self._loading_overlay_locator).is_displayed())
+        except NoSuchElementException:
+            pass
+        Wait(self.marionette).until(expected.element_displayed(
+            Wait(self.marionette).until(expected.element_present(
+                *self._section_wifi_locator))))
```

Comment:
Yeah, it was introduced later. 

Dave Hunt

> On 23 Aug 2015, at 09:08, Martijn <notifications@github.com> wrote:
> 
> In tests/python/gaia-ui-tests/gaiatest/apps/ftu/app.py:
> 
> > @@ -159,8 +159,14 @@ def tap_next_to_wifi_section(self):
> >  
> >      def a11y_click_next_to_wifi_section(self):
> >          self.a11y_click_next()
> > -        self.wait_for_element_not_displayed(*self._loading_overlay_locator)
> > -        self.wait_for_element_displayed(*self._section_wifi_locator)
> > +        try:
> > +            Wait(self.marionette, timeout, ignored_exceptions=StaleElementException).until(
> > +                lambda m: not m.find_element(*self._loading_overlay_locator).is_displayed())
> > +        except NoSuchElementException:
> > +            pass
> > +        Wait(self.marionette).until(expected.element_displayed(
> > +            Wait(self.marionette).until(expected.element_present(
> > +                *self._section_wifi_locator))))
> Ah, wait, from a pull request comment in bug 1074117:
> "
> element_displayed can now take a locator, so this can be written as:
> 
> Wait(self.marionette).until(expected.element_displayed(*self._status_bar_locator)
> "
> 
> So I guess we should just use this wherever we can, instead.
> 
> —
> Reply to this email directly or view it on GitHub.
>